shared/src/commonTest/kotlin/org/hyperskill/step/StepRouteTest.kt

```diff
@@ -0,0 +1,45 @@
+package org.hyperskill.step
+
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import org.hyperskill.app.step.domain.model.StepRoute
+import org.hyperskill.app.step.domain.model.copy
+
+class StepRouteTest {
+    @Test
+    fun `StepRoute copy with new step id`() {
+        val initialStepId = 1L
+        val initialStepRoutes = listOf(
+            StepRoute.Learn(initialStepId),
+            StepRoute.LearnDaily(initialStepId),
+            StepRoute.Repeat(initialStepId),
+            StepRoute.StageImplement(initialStepId, 2L, 3L),
+        )
+
+        val expectedStepId = 2L
+        val expectedStepRoutes = listOf(
+            StepRoute.Learn(expectedStepId),
+            StepRoute.LearnDaily(expectedStepId),
+            StepRoute.Repeat(expectedStepId),
+            StepRoute.StageImplement(expectedStepId, 2L, 3L),
+        )
+
+        initialStepRoutes.forEachIndexed { index, initialStepRoute ->
+            val expectedStepRoute = expectedStepRoutes[index]
+            val actualStepRoute = initialStepRoute.copy(expectedStepId)
+
+            assertEquals(expectedStepRoute::class, actualStepRoute::class)
+            assertEquals(expectedStepRoute.stepId, actualStepRoute.stepId)
+            assertEquals(expectedStepRoute.analyticRoute.path, actualStepRoute.analyticRoute.path)
+            assertEquals(expectedStepRoute.stepContext, actualStepRoute.stepContext)
+
+            if (expectedStepRoute is StepRoute.StageImplement) {
+                assertEquals(
+                    expectedStepRoute.projectId,
+                    (actualStepRoute as StepRoute.StageImplement).projectId
+                )
+                assertEquals(expectedStepRoute.stageId, actualStepRoute.stageId)
+            }
```

Comment:
You can mark all the stepRoot classes as `data` or just to implement custom `equals` methods to compare them.
Then you can replace fields by fields assertion with just a `assertEquals(expectedStepRoute, actualStepRoute)`

Reply:
It is marked as data classes instead of custom equals because we can forget to update equals implementation if smth is changed.